### PR TITLE
W7 - hieucap - UI test

### PR DESCRIPTION
**Pull request by at-hieucap**
#### Related issues (Optional)
Task #84 

#### Description (Required)
- [x] Create UI test

### CODE REVIEW

File: app/src/androidTestAt-hieucapDebug/java/LoginUITest.java [hunk 1]
```diff
@@ -0,0 +1,228 @@
+import android.support.test.rule.ActivityTestRule;
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-hieucapDebug/java/LoginUITest.java]: WARNING: [TooManyStaticImports] Too many static imports may lead to messy code
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#TooManyStaticImports

File: app/src/androidTestAt-hieucapDebug/java/LoginUITest.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.StringStartsWith.startsWith;
+
+/**
+ * Create UI test
+ * Created by tiboo on 05/01/2018.
+ */
+@RunWith(AndroidJUnit4.class)
+public class LoginUITest {
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-hieucapDebug/java/LoginUITest.java]: WARNING: [NoPackage] All classes and interfaces must belong to a named package
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#NoPackage

File: app/src/androidTestAt-hieucapDebug/java/LoginUITest.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        onView(withText(startsWith("Password is empty")))
+                .inRoot(withDecorView(not(is(mActivityRule.getActivity().getWindow().getDecorView()))))
+                .check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void checkNullPassword() {
+        // Type text username and then press the button.
+        onView(withId(R.id.edtUserName))
+                .perform(typeText("Hieu22intern"), closeSoftKeyboard());
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-hieucapDebug/java/LoginUITest.java]: WARNING: [AvoidDuplicateLiterals] The String literal "Hieu22intern" appears 9 times in this file; the first occurrence is on line 120
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/at-hieucap/java/vn/asiantech/internship/unit_test/PasswordValidationTest.java [hunk 4]
```diff
@@ -33,8 +30,11 @@ public static boolean isSameCharacter(String password) {
                     count++;
                 }
             }
+            if (count >= 2) {
+                return false;
+            } else count = 0;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/unit_test/PasswordValidationTest.java]: WARNING: [IfElseStmtsMustUseBraces] Avoid using if...else statements without curly braces
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/braces.html#IfElseStmtsMustUseBraces

File: app/src/androidTestAt-hieucapDebug/java/LoginUITest.java [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.StringStartsWith.startsWith;
+
+/**
+ * Create  login UI test
+ * Created by tiboo on 05/01/2018.
+ */
+@RunWith(AndroidJUnit4.class)
+public class LoginUITest {
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-hieucapDebug/java/LoginUITest.java]: WARNING: [NoPackage] All classes and interfaces must belong to a named package
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#NoPackage

**Comment by at-hieucap** [comment]
@at-trungnguyen  I finished exercise , please review for me,
Thank you so much !!!

**Comment by at-trungnguyen** [comment]
@at-hieucap Please fix conflict!

File: app/src/androidTestAt-hieucapDebug/java/LoginUITest.java [hunk 6]
```diff
@@ -0,0 +1,229 @@
+import android.support.test.rule.ActivityTestRule;
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-hieucapDebug/java/LoginUITest.java]: WARNING: [TooManyStaticImports] Too many static imports may lead to messy code
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#TooManyStaticImports

File: app/src/androidTestAt-hieucapDebug/java/LoginUITest.java (hunk 5 above)
Comment (at-dev-support-01) [inline app/src/androidTestAt-hieucapDebug/java/LoginUITest.java]: WARNING: [NoPackage] All classes and interfaces must belong to a named package
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#NoPackage

**Comment by at-hieucap** [comment]
@at-trungnguyen I fixed conflict , please review for me,
Thank you so much !!!

**Pull request merged**